bench/collatz/CMakeLists.txt

```diff
@@ -23,6 +23,11 @@ add_exe(collatz-profiling-shared
         LIBRARIES Threads::Threads dd_profiling-shared
         DEFINITIONS ${COLLATZ_DEFINITION_LIST} USE_DD_PROFILING)
 target_include_directories(collatz-profiling-shared PRIVATE ${STATSD_INCLUDE_DIR})
+# Fix for link error in sanitizeddebug build mode with gcc:
+# /usr/bin/ld: ./libtest.so: undefined reference to `__dynamic_cast'
+# /usr/bin/ld: ./libtest.so: undefined reference to `typeinfo for __cxxabiv1::__vmi_class_type_info'
+# Workaround is to link against libstdc++
+target_link_libraries(collatz-profiling-shared PRIVATE "$<$<AND:$<C_COMPILER_ID:GNU>,$<CONFIG:SanitizedDebug>>:-Wl,-Bstatic;-lstdc++;-Wl,-Bdynamic>")
```

Comment:
IIRC, clang respects `-lstdc++` just fine, since you have to give it `-stdlib=libc++` in order to make it use the llvm C++ runtime in the first place.  On the other hand, your note makes it clear that this is only an issue on GCC (no surprise!).

Would it simplify things to emit the linker command regardless of compiler?  It shouldn't create a new problem for Clang.

Slightly doesn't matter either way, since this is just a test app anyway.  :)